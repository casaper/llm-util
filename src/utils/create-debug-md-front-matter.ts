import { ChatResponse } from 'ollama';

import { msToHumanDuration } from './ms-to-human-duration';

export const createDebugMdFrontMatter = ({
  model,
  eval_count,
  prompt_eval_count,
  total_duration,
  eval_duration,
  load_duration,
  prompt_eval_duration,
  created_at,
}: ChatResponse) =>
  [
    '---',
    `model: ${model}`,
    `eval_count: ${eval_count}`,
    `prompt_eval_count: ${prompt_eval_count}`,
    `total_duration: ${msToHumanDuration(total_duration)}`,
    `eval_duration: ${msToHumanDuration(eval_duration)}`,
    `load_duration: ${msToHumanDuration(load_duration)}`,
    `prompt_eval_duration: ${msToHumanDuration(prompt_eval_duration)}`,
    `created_at: ${created_at.toISOString()}`,
    '---',
    '',
    '',
  ].join('\n');
