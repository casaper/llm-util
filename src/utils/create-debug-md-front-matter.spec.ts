import { ChatResponse } from 'ollama';
import { describe, expect, it } from 'vitest';

import { createDebugMdFrontMatter } from './create-debug-md-front-matter';

describe('createDebugMdFrontMatter', () => {
  const mockChatResponse = {
    model: 'llama3.1:8b',
    created_at: new Date('2024-01-01T00:00:00Z'),
    eval_count: 100,
    prompt_eval_count: 200,
    total_duration: 123400,
    eval_duration: 56700,
    load_duration: 89000,
    prompt_eval_duration: 12300,
    message: {
      role: 'assistant',
      content: 'This is a test response.',
    },
    done: true,
    done_reason: 'done',
  } satisfies ChatResponse;

  it('should generate correct front matter', () => {
    const result = createDebugMdFrontMatter(mockChatResponse);
    expect(result).toMatchSnapshot();
  });

  it('should handle zero durations correctly', () => {
    const responseWithZeroDurations = {
      ...mockChatResponse,
      total_duration: 0,
      eval_duration: 0,
      load_duration: 0,
      prompt_eval_duration: 0,
    };
    const result = createDebugMdFrontMatter(responseWithZeroDurations);
    expect(result).toMatchSnapshot();
  });
});
