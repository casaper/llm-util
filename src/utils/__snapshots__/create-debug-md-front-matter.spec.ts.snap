// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`createDebugMdFrontMatter > should generate correct front matter 1`] = `
"---
model: llama3.1:8b
eval_count: 100
prompt_eval_count: 200
total_duration: 2m 3s 400ms
eval_duration: 56s 700ms
load_duration: 1m 29s
prompt_eval_duration: 12s 300ms
created_at: 2024-01-01T00:00:00.000Z
---

"
`;

exports[`createDebugMdFrontMatter > should handle zero durations correctly 1`] = `
"---
model: llama3.1:8b
eval_count: 100
prompt_eval_count: 200
total_duration: 0s
eval_duration: 0s
load_duration: 0s
prompt_eval_duration: 0s
created_at: 2024-01-01T00:00:00.000Z
---

"
`;
